crates/rpc/rpc-eth-api/src/helpers/evm_tracer.rs

```diff
@@ -0,0 +1,126 @@
+//! EVM Tracer for executing transactions with a custom inspector and handling results.
+use alloy_rpc_types_eth::TransactionInfo;
+use reth_evm::{Evm, EvmEnv, EvmFactory, IntoTxEnv};
+use revm::{
+    context::TxEnv, context_interface::result::ResultAndState, interpreter::InterpreterTypes,
+    primitives::hardfork::SpecId, Database, Inspector,
+};
+use std::{fmt, marker::PhantomData};
+
+/// A struct that represents an EVM tracer for executing transactions with a custom inspector.
+pub struct EvmTracer<E, Txs, I, DB, H>
+where
+    E: EvmFactory,
+    DB: InterpreterTypes + Database + Send + Sync + 'static,
```

Comment:
I think we should try restricting this to just `trait Evm`:

https://github.com/alloy-rs/evm/blob/f3ea0bc563c4259aa9c02288e6ec70fc438c926d/crates/evm/src/evm.rs#L26-L26

so that the user is responsible for providing an Evm instance